src/main/scala/viper/gobra/frontend/Config.scala

```diff
@@ -16,22 +16,37 @@ import org.rogach.scallop.{ScallopConf, ScallopOption, listArgConverter, singleA
 import org.slf4j.LoggerFactory
 import viper.gobra.backend.{ViperBackend, ViperBackends}
 import viper.gobra.GoVerifier
-import viper.gobra.frontend.PackageResolver.RegularImport
+import viper.gobra.frontend.Source.FromFileSource
 import viper.gobra.reporting.{FileWriterReporter, GobraReporter, StdIOReporter}
 import viper.gobra.util.{TypeBounds, Violation}
 import viper.silver.ast.SourcePosition
 
+import scala.concurrent.duration.Duration
+
 object LoggerDefaults {
   val DefaultLevel: Level = Level.INFO
 }
+
+object ConfigDefaults {
+  val DefaultGobraDirectory: String = ".gobra"
+  val DefaultTaskName: String = "gobra-task"
+}
+
 case class Config(
                  inputs: Vector[Source],
+                 recursive: Boolean = false,
+                 gobraDirectory: Path = Path.of(ConfigDefaults.DefaultGobraDirectory),
+                 // Used as a unique identifier per verification task, ideally it shouldn't change between verifications
+                 // because it is used as a caching key, additionally
+                 taskName: String = ConfigDefaults.DefaultTaskName,
```

Comment:
How do you ensure that different verification tasks use different task names? You pass a single `config` instance to your verify all packages functions. Doesn't this then result in all of them using the same task name?

Reply:
`verifyAllPackages` loops over the `inputPackageMap` of the config, where we grouped the input files by `packageId`. It then copies config and overwrites `inputs`, `taskName` and `reporter` fields. It uses the `packageId` as a taskName which should be unique. Otherwise, I don't provide any guarantees